public/app/features/logs/logsModel.ts

```diff
@@ -457,6 +457,11 @@ export function logSeriesToLogsModel(logSeries: DataFrame[], queries: DataQuery[
     }
   }
 
+  // Until nanosecond precision for requests is supported, we need to filter possible duplicates
+  if (config.featureToggles.logsInfiniteScrolling) {
```

Comment:
@svennergr Of all the possible places I considered to remove duplication, this was the one that made the most sense and had no visual impact, because here not only we process and return log rows, but also calculate meta stats. Without this, there was a discrepancy between meta stats and the displayed data. Let me know what you think!

Reply:
I like that a lot. One possible improvement could be to move that logic to line ~456 and just call `rows.push(row);` if `rows` doesn't include `row`?
With that we wouldn't need to iterate through all the fields twice.

Reply:
Makes a lot of sense. I'm also thinking of using a slightly modified `refId` for infinite scroll queries, so we are extra sure we're not disrupting other normal behaviors.

Reply:
Either that or passing a new parameter to `logSeriesToLogsModel` that is only set when used from infinite scrolling. However, the `refId` approach also sounds reasonable for now.

Reply:
We now use a special refId for infinite scrolling requests, similar to supplementary queries, so it will not trigger deduplication unless this specific use case.